src/odemis/gui/comp/legend.py

```diff
@@ -245,16 +250,15 @@ def set_stream_type(self, side, acq_type):
         :param acq_type: (String): acquisition type associated with stream
         """
 
-        for t, type_icon in self._type_to_icon:
-            if acq_type == t:
-                icon = type_icon
-                break
-        else:
+        try:
+            icon = self._type_to_icon[acq_type]
+        except KeyError as ex:
             # Don't fail too bad
             icon = img.getBitmap("icon/ico_blending_opt.png")
             if self.merge_slider.IsShown():
-                logging.warning("Failed to find icon for stream of type %s",
-                                acq_type)
+                # For KeyError, the "message" is just the key name
+                logging.warning("Failed to find icon for stream of type %s", ex)
```

Comment:
This feels like you're mis-using the `KeyError`, why not do something like:
```python
        icon = self._type_to_icon.get(acq_type, None)
        if icon is None:
            # Don't fail too bad
            icon = img.getBitmap("icon/ico_blending_opt.png")
            if self.merge_slider.IsShown():
                logging.warning("Failed to find icon for stream of type %s",
                                acq_type)
```

Reply:
I disagree it's a mis-use of Exceptions. I'd even argue it's precisely what Exceptions are intended for! The normal code is in the `try` block. We expect that this code works. However, if for some reason we haven't handled it properly (eg, new stream type not yet added), we fallback gracefully in the except block.
Admittedly, it's a thin line. Especially in this code, I'm aware that there are cases where the stream is unknown... That should be only temporary though, while we make the icons for these streams.

Reply:
Okay, I would say that at least according to realpython the usual solution is to use `.get()`, but your way is at least the general solution. For readability I still prefer `logging.warning("Failed to find icon for stream of type %s", acq_type)` over `logging.warning("Failed to find icon for stream of type %s", ex)`.

Reply:
Oh, indeed, using `acq_type` is much nicer... I don't know how, but I got convinced that it wasn't available and I had to take the info from the exception!